src/app/EventManagement.cpp

```diff
@@ -748,6 +751,71 @@ CHIP_ERROR EventManagement::FetchEventsSince(TLVWriter & aWriter, const ObjectLi
     return err;
 }
 
+CHIP_ERROR EventManagement::RemoveInvalidFabricCB(const TLV::TLVReader & aReader, size_t aDepth, void * apContext)
+{
+    // the function does not actually remove the event, instead, it sets the fabric index to an invalid value.
+    FabricIndex * invalidFabricIndex = static_cast<FabricIndex *>(apContext);
+
+    TLVReader event;
+    TLVType tlvType;
+    TLVType tlvType1;
+    event.Init(aReader);
+    ReturnErrorOnFailure(event.EnterContainer(tlvType));
+    ReturnErrorOnFailure(event.Next());
+    ReturnErrorOnFailure(event.EnterContainer(tlvType1));
+    ReturnErrorOnFailure(event.Next());
+
+    uint8_t fabricIndex = 0;
+    while (CHIP_NO_ERROR == event.Next())
+    {
+        if (event.GetTag() == TLV::ProfileTag(kEventManagementProfile, kFabricIndexTag))
+        {
+            ReturnErrorOnFailure(event.Get(fabricIndex));
+            if (fabricIndex == *invalidFabricIndex)
+            {
+                // fabricIndex is encoded as an integer; the dataPtr will point to a location immediately after its encoding
+                uint8_t * dataPtr                  = const_cast<uint8_t *>(event.GetReadPoint());
+                CHIPCircularTLVBuffer * readBuffer = static_cast<CHIPCircularTLVBuffer *>(event.GetBackingStore());
+                // shift the dataPtr to point to the encoding of the fabric index, accounting for wraparound in backing storage
+                if (readBuffer->GetQueue() != dataPtr)
+                {
+                    dataPtr = dataPtr - 1;
```

Comment:
This is assuming min-length encoding.  Should at least document that, if we can't check what the actual encoding size is...

Reply:
documented, thanks